wbox_configure.sh

```diff
@@ -1,9 +1,9 @@
 #!/bin/sh
 set -e
 if [ -z "$SYSROOT" ]; then export SYSROOT="$(realpath "$(dirname "$0")/../sysroot")"; fi
+if [ -z "$CC" ]; then export CC="clang"; fi
 
-export LDFLAGS=""
-export CFLAGS="-Werror -fvisibility=hidden -mcmodel=large -mstack-protector-guard=global -no-pie -fno-pic -fno-pie" # TODO: Add -flto
-export AR="gcc-ar"
-export RANLIB="gcc-ranlib"
+export CFLAGS="-Werror -fvisibility=hidden -mcmodel=large -mstack-protector-guard=global -fno-pic -fno-pie -fcf-protection=none" # TODO: Add -flto
+export AR="llvm-ar"
+export RANLIB="llvm-ranlib"
```

Comment:
Could this be switched by user `CC` choice as well? Aka using gcc-ar for gcc and llvm-ar for clang? Or is there a reason linking is now always done with llvm?

Reply:
-ar and -ranlib really don't matter in any case, they just archive the objects into .a files. llvm-ar and llvm-ranlib would be more "guaranteed" to exist given the waterbox setup requires llvm package to be installed. Although I suppose at that point just let autotools figure out what ar/ranlib is available.

Reply:
Fwiw on my ubuntu install, just installing `clang` installs `llvm-15` as a dependency, however that only provides `llvm-ar-15` (etc.). I did not have the `llvm` package explicitly installed, which would have provided those symlinks (which is arguably my fault because the readme does list `llvm` as a requirement).